Write up Content section (#37)
directus/docs: content/3.content/2.editor.md

---
description: Learn to create, duplicate, archive and perform other actions with items using Directus.
---

# Item Editor

> The item editor is a tailored form for managing items and their field values.

Reviewer: Maybe mention it's specifically a SINGLE item?

<!-- TODO: IMAGE OF EDITOR -->

## Fields & Data Model

You can add fields to items by [configuring the collection's data model](/data-modeling/fields). Here, you can also configure how the fields are displayed in the item editor.

## Creating Items

> To create an Item, navigate to the content module and select the desired collection. Click :icon{name="material-symbols:add-circle-outline-rounded"} in the page header to open the item page. Fill in item details as desired. Click :icon{name="material-symbols:check"} in the page header.

Reviewer: Lowercase item please. This is a difficult section - is it too patronising? What else matters here? Validation maybe? Conditionally-shown fields?

::callout{type="info" title="Singletons"}

If the collection is configured as a [singleton](/data-modeling/collections) in the data model

> settings, the App will automatically open the item page when selecting said collection.

Reviewer: ```suggestion
If the collection is configured as a [singleton](/data-modeling/collections) in the data model
settings, the App will automatically open the item page when selecting the collection.
```

Reviewer: I'd also maybe add that singletons are typically used as if they were a key-value store, even if it is really a collection with only a single item?

::

## Duplicating Items

<!-- TODO: IMAGE -->

> When editing an item, you can click on :icon{name="material-symbols:more-vert"} to select some advanced options, amongst them "Save as Copy". Selecting this will save said copy.

Reviewer: ```suggestion
When editing an item, you can click on :icon{name="material-symbols:more-vert"} to select some advanced options, amongst them "Save as Copy". Selecting this will save a copy.
```

You'll notice this a bunch. I think 'a' or 'the' is almost always better than 'said', which feels a bit academic.

<!-- Save as copy when small change -->
<!-- TODO: IMAGE -->

When editing an item, you can click on :icon{name="material-symbols:more-vert"} to select some advanced options, amongst them "Save as Copy". Selecting this will save said copy.

## Archiving Items

> <!-- 'soft delete' -->

Reviewer: Can you talk about this being like a 'soft delete'. In that, by default, it won't show up in search results or be returned by the API. But they still exist and, given the correct queries, will be accessible.

> Requires an [archive field](/data-modeling/collections) to be configured within the collection's data model

Reviewer: ```suggestion
Archiving requires an [archive field](/data-modeling/collections) to be configured within the collection's data model
```

::

## Revisions

<!-- TODO: IMAGE -->

As you update field values on items, Directus saves these revisions, and they can be compared side-by-side to the current state.

> To revert an item, navigate to the content module and select the desired collection and select the desired item. Click on "Revisions" in the page sidebar and then on the revision you wish to preview. Go to "Revisions Made" in the side menu and view the revision differences. Click :icon{name="material-symbols:settings-backup-restore"} to revert the item's values and return to the item Page.

Reviewer: ```suggestion
To revert an item, navigate to the content module and select the desired collection and select the desired item. Click on "Revisions" in the editor sidebar and then on the revision you wish to preview. Go to "Revisions Made" in the side menu and view the revision differences. Click :icon{name="material-symbols:settings-backup-restore"} to revert the item's values and return to the item page.
```

::callout{type="info" title="Revision Preview"}

You will also see a "Revision Preview" button in the side menu navigation, which will let you preview all the item's
values for that revision.

::

::callout{type="info"}

> You can also revert Items [programmatically via the API](/api-reference/system/revisions).

Reviewer: ```suggestion
You can also revert items [programmatically via the API](/api-reference/system/revisions).
```

:::

## Comments

<!-- TODO: IMAGE -->

> You can add comments to items in the sidebar by clicking on "Comments", which will show the form for submitting one. You can use the @ button to tag specific users in said comment.

Reviewer: ```suggestion
You can add comments to items in the sidebar by clicking on "Comments", which will show the form for submitting one. You can use the @ button to tag specific users in your comment.
```
